### PR TITLE
Pull request

### PR DESCRIPTION
**Pull request by DenysLutskiy**

### CODE REVIEW

File: shema.gql [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+type Query {
+  getItems(take: Int, skip: Int): [Item]!
+  searchWithFilter(filter: ItemFilter, take: Int, skip: Int): [SearchResult]!
+  getUserByName(username: String!): User
+}
+
+type Mutation {
+  signupUser(user: CreateUserInput!): RegisterResponse!
+  loginUser(user: LoginUserInput!): LoginResponse!
+  createItem(item: CreateItemInput!): Item!
```
Comment (d-smyrnov) [inline shema.gql:12]: Item-related mutations could also have separate types.

File: shema.gql [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+# ---------------------------------- Filters --------------------------------- #
+# ---------------------------------------------------------------------------- #
+input ItemFilter {
+  containsInName: String
+  containsInDescription: String
+  containsInDisount: String
+}
+
+# ----------------------- Interfaces and Responce types ---------------------- #
```
Comment (d-smyrnov) [inline shema.gql:64]: Reponce - check for typo

File: shema.gql [hunk 3]
```diff
@@ -0,0 +1,179 @@
+# --------------------------- Querys and Mutations --------------------------- #
+# ---------------------------------------------------------------------------- #
+type Query {
+  getItems(take: Int, skip: Int): [Item]!
```
Comment (d-smyrnov) [inline shema.gql:4]: Recheck also other array return types (relating to https://futurotech.atlassian.net/browse/FT-36?focusedCommentId=17107 comments). Update where you don't expect arrays to have null items.

File: shema.gql [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+  duration: PeriodOfTime!
+  assignedTo: [Item]!
+  createdAt: Date!
+}
+
+# ----------------------------- Enums and scalars ---------------------------- #
+# ---------------------------------------------------------------------------- #
+scalar Date
+
+scalar PeriodOfTime
```
Comment (d-smyrnov) [inline shema.gql:147]: You can add comments to describe what format that scalar type represents and what is it used for. 
```suggestion
"..."
scalar Date

"..."
scalar PeriodOfTime
```

http://spec.graphql.org/draft/#sec-Descriptions

File: shema.gql (hunk 3 above)
Comment (denismukhopadov) [inline shema.gql:4]: Also you can use type for pagination filter, because you are using it many times

File: shema.gql (hunk 1 above)
Comment (DenysLutskiy) [reply shema.gql:12]: Could you clarify your comment, please? 
You meant that I should create Response type for every mutation or it's just recommendation?

File: shema.gql [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+  errors: [Error]
+  user: Comment!
+}
+
+union SearchResult = User | Item
+
+# ------------------------------- Object types ------------------------------- #
+# ---------------------------------------------------------------------------- #
+type User {
+  _id: ID!
```
Comment (denismukhopadov) [inline shema.gql:93]: you can name this field id instead of _id like its in mongo, i think object will be transormed

File: shema.gql (hunk 3 above)
Comment (DenysLutskiy) [reply shema.gql:4]: Am I understand right. that I must use [Any**!**]**!** format, always when I waiting for array type in response?

File: shema.gql (hunk 1 above)
Comment (denismukhopadov) [reply shema.gql:12]: @DenysLutskiy He means that you have to create CreateItemResponse type that may be differ from default Item entity

Comment (d-smyrnov) [reply shema.gql:12]: it's not a strict rule, just a recommendation that can be omitted 😁

File: shema.gql (hunk 3 above)
Comment (d-smyrnov) [reply shema.gql:4]: If you don't expect to have null values in the array - yes.

**Pull request closed without merging**